lib/pure/asyncdispatch.nim

```diff
@@ -1484,16 +1505,26 @@ else:
     addRead(socket, cb)
     return retFuture
 
-  proc acceptAddr*(socket: AsyncFD, flags = {SocketFlag.SafeDisconn}):
+  proc acceptAddr*(socket: AsyncFD, flags = {SocketFlag.SafeDisconn},
+                   inheritable = defined(nimInheritHandles)):
       owned(Future[tuple[address: string, client: AsyncFD]]) =
     var retFuture = newFuture[tuple[address: string,
         client: AsyncFD]]("acceptAddr")
     proc cb(sock: AsyncFD): bool =
       result = true
       var sockAddress: Sockaddr_storage
       var addrLen = sizeof(sockAddress).SockLen
-      var client = accept(sock.SocketHandle,
-                          cast[ptr SockAddr](addr(sockAddress)), addr(addrLen))
+      var client =
+        when declared(accept4):
```

Comment:
Since `accept4` is only available in newer Linux kernels, let's make it possible to disable it's usage using this flag:

```suggestion
        when declared(accept4) and defined(nimInheritHandles):
```

Reply:
I'd say kernel 2.6.28 is old enough :P, but sure, I'll add something to disable `accept4` itself from `posix`.

Reply:
2008 is indeed old, but if we can support the previous versions easily we should. You never know who might need it.

Reply:
I'll make some changes to facilitate introducing a new flag: `nimOldPosix` for when these newer procs are not supported.

But adding the flag itself will have to be done in an another PR, since it'd touch many other parts of the stdlib.